samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoAvailableNotificationServiceImpl.java

```diff
@@ -0,0 +1,373 @@
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.samigo.api.SamigoAvailableNotificationService;
+import org.sakaiproject.samigo.util.SamigoConstants;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.time.api.UserTimeService;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.assessment.data.dao.assessment.ExtendedTime;
+import org.sakaiproject.tool.assessment.data.ifc.assessment.AssessmentBaseIfc;
+import org.sakaiproject.tool.assessment.facade.PublishedAssessmentFacade;
+import org.sakaiproject.tool.assessment.services.PersistenceService;
+import org.sakaiproject.tool.assessment.services.assessment.PublishedAssessmentService;
+import org.sakaiproject.tool.cover.ToolManager;
+import org.sakaiproject.user.api.Preferences;
+import org.sakaiproject.user.api.PreferencesService;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+import org.sakaiproject.util.ResourceLoader;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Slf4j
+public class SamigoAvailableNotificationServiceImpl implements SamigoAvailableNotificationService {
+
+    @Setter private EmailService emailService;
+    @Setter private PreferencesService preferencesService;
+    @Setter private ScheduledInvocationManager scheduledInvocationManager;
+    @Setter private ServerConfigurationService serverConfigurationService;
+    @Setter private SessionManager sessionManager;
+    @Setter private SiteService siteService;
+    @Setter private UserDirectoryService userDirectoryService;
+    @Setter private UserTimeService userTimeService;
+    private PublishedAssessmentService publishedAssessmentService;
+    private static final ResourceLoader rl = new ResourceLoader("SamigoAvailableNotificationMessages");
+    private final List<String> additionalHeaders = new ArrayList<>();
+    //feedback
+    public static Integer NO_FEEDBACK = Integer.valueOf(3);
+    //scoring types
+    public static final Integer HIGHEST_SCORE = 1;
+    public static final Integer LAST_SCORE= 2;
+    public static final Integer ALL_SCORE= 3;
+    public static final Integer AVERAGE_SCORE= 4;
+    //this matches DEAD_STATUS from AssessmentBaseIfc, indicating a deleted assessment
+    public static final Integer DEAD_STATUS = 2;
+
+    public void init() {
+        log.debug("SamigoAvailableNotificationService init()");
+
+        publishedAssessmentService = new PublishedAssessmentService();
+
+        String sender = "Sender: \"" + getServiceName() + "\" <" + getSetupRequest() + ">";
+        additionalHeaders.add(sender);
+        additionalHeaders.add("Content-type: text/html; charset=UTF-8");
+    }
+
+    public void destroy() {
+        log.debug("SamigoAvailableNotificationService destroy()");
+    }
```

Comment:
Not neeeded

Reply:
I went ahead and pushed this change so you could review again, but I had some issues starting Tomcat with this removed in my local testing. It compiled fine but Tomcat just didn't start correctly, with an error in the logs specifically calling out the absence of this method.

Reply:
Never mind...based on a suggestion from @davidpbauer , I was able to make this change work. Mention of this method also had to be removed from the relevant components.xml file.